packages/js/product-editor/src/components/block-editor/block-editor.tsx

```diff
@@ -182,14 +182,16 @@ export function BlockEditor( {
 		};
 	}, [ settingsGlobal ] );
 
-	const { editedRecord: product, hasResolved } = useEntityRecord< Product >(
+	const { editedRecord: product } = useEntityRecord< Product >(
 		'postType',
 		postType,
 		productId,
 		// Only perform the query when the productId is valid.
 		{ enabled: productId !== -1 }
 	);
 
+	const hasResolved = !! product;
+
```

Comment:
This way triggering `invalidateResolution` doesn't trigger a full refresh.
But an alternative would be addressing this: https://github.com/woocommerce/woocommerce/blob/2db29164f9d9baf61aecf94e06b1658a57bf866a/packages/js/product-editor/src/hooks/use-product-variations-helper.ts#L163-L166